app/bundles/LeadBundle/Event/DoNotContactAddEvent.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Mautic\LeadBundle\Event;
+
+use Mautic\LeadBundle\Entity\DoNotContact as DNC;
+use Symfony\Component\EventDispatcher\Event;
+
+class DoNotContactAddEvent extends Event
+{
+    public const ADD_DONOT_CONTACT = 'mautic.lead.add_donot_contact';
+
+    private $lead;
+
+    private $channel;
+
+    private $comments;
+
+    private $reason;
+
+    private $persist;
+
+    private $checkCurrentStatus;
+
+    private $override;
+
+    public function __construct($lead, $channel, $comments = '', $reason = DNC::BOUNCED, $persist = true, $checkCurrentStatus = true, $override = true)
```

Comment:
```suggestion
    public function __construct(Lead $lead, string $channel, string $comments = '', string $reason = DNC::BOUNCED, bool $persist = true, bool $checkCurrentStatus = true, bool $override = true)
```
Since we have minimal PHP version of 7.2 for M3, let's use all goodies we can in the newly added classes and methods.